src/main/haskell/kore/test/Test/Kore.hs

```diff
@@ -125,108 +125,141 @@ moduleNameGen :: Gen ModuleName
 moduleNameGen = ModuleName <$>
     genericIdGen idFirstChars (idFirstChars ++ idOtherChars)
 
-variableGen :: MetaOrObject level => level -> Gen (Variable level)
-variableGen x = pure Variable
-    <*> scale (`div` 2) (idGen x)
-    <*> scale (`div` 2) (sortGen x)
+variableGen
+    :: MetaOrObject level
+    => [Variable level]
+    -> level
+    -> Gen (Variable level)
+variableGen vars x = oneof (newVar : if null vars then [] else [elements vars])
```

Comment:
Why not this?
```suggestion
variableGen vars x = oneof (newVar : map elements vars)
```

Reply:
Yeah, I think I meant `map pure vars`. With the original, there is a 50% chance to introduce a new variable, instead of a uniform probability to introduce a new variable or re-use a bound variable. I'm not sure which distribution is better for these tests.